library/IvozProvider/Model/Companies.php

```diff
@@ -184,12 +190,16 @@ public function setDomainUsers($data)
      */
     public function getDomain()
     {
-        $domains = $this->getDomains();
-        if (!empty($domains)) {
-            $domain = array_shift($domains);
-            return $domain->getDomain();
+        if ($this->getType() === Companies::RETAIL) {
+            return $this->getBrand()->getDomainUsers();
         } else {
-            return "";
+            $domains = $this->getDomains();
+            if (!empty($domains)) {
+                $domain = array_shift($domains);
+                return $domain->getDomain();
+            } else {
+                return "";
```

Comment:
In fact, there should be a method called getDomainUsers that does exactly the same as the whole firat else.